### PR TITLE
Better iPhone metrics

### PR DESCRIPTION
**Pull request by gklopper**
This takes into account (some of) the different iOs versions.

### CODE REVIEW

File: common/app/views/fragments/javaScriptFirstSteps.scala.html [hunk 1]
```diff
@@ -194,35 +194,42 @@
 
 </script>
 
-
-<script>
-    guardian.isIphone6 = false;
-    guardian.isIphone4 = false;
-    guardian.inTestBucket = Math.floor(Math.random() * 2 + 1) === 1;
```
Comment (gklopper) [inline common/app/views/fragments/javaScriptFirstSteps.scala.html]: Goodbye embarrassing code

File: common/app/views/fragments/javaScriptFirstSteps.scala.html [hunk 2]
```diff
@@ -194,35 +194,42 @@
...
-            guardian.isIphone6 = (screen.width === 375 && screen.height === 667) || (screen.width === 414 && screen.height === 736);
-            guardian.isIphone4 = screen.width === 320 && screen.height === 480 && window.devicePixelRatio > 1;
+            var iPhoneVersionRegex = /.*iPhone OS ([678])_\d+/;
 
-            if (guardian.isIphone6) {
-                (new Image()).src = guardian.config.page.beaconUrl + '/count/iphone-6-start.gif';
-            }
+            if (iPhoneVersionRegex.test(navigator.userAgent)) {
+
+                var osVersion = iPhoneVersionRegex.exec(navigator.userAgent)[1];
```
Comment (phamann) [inline common/app/views/fragments/javaScriptFirstSteps.scala.html]: What if the exec returns null, will you get a runtime exception for trying to access and array?

**Comment by phamann** [comment]
:+1: Other than comment

File: common/app/views/fragments/javaScriptFirstSteps.scala.html (hunk 2 above)
Comment (gklopper) [inline common/app/views/fragments/javaScriptFirstSteps.scala.html]: It only gets that far if it passes line 221 `iPhoneVersionRegex.test(navigator.userAgent)` i.e. it matches the pattern.

**Pull request merged**